### PR TITLE
OperatorContextGet[Double, Int]

### LINKED ISSUE #1144: Add `CeedOperatorContextGetDouble` and friends for label data access
A few questions for `CeedOperatorContextGetDouble` would be:
1) Is getting data access to the label expensive? 
   - ie. should we add a member to `CeedContextFieldLabel` that keeps track of the value to avoid trying to get access to the context (thinking GPU to CPU communication might be at play here)
2) For sublabels, do we trust that all the contexts will be the same value and, if not, what value do we return?

### PR DESCRIPTION
**Pull request by jeremylt**
- [x] Add Context[Get,Restore][Double,Int32]Read
- [x] Add tests

Closes #1144

### CODE REVIEW

File: include/ceed/ceed.h [hunk 1]
```diff
@@ -508,6 +508,8 @@ CEED_EXTERN int CeedOperatorGetNumElements(CeedOperator op, CeedInt *num_elem);
 CEED_EXTERN int CeedOperatorGetNumQuadraturePoints(CeedOperator op, CeedInt *num_qpts);
 CEED_EXTERN int CeedOperatorGetFlopsEstimate(CeedOperator op, CeedSize *flops);
 CEED_EXTERN int CeedOperatorContextGetFieldLabel(CeedOperator op, const char *field_name, CeedContextFieldLabel *field_label);
+CEED_EXTERN int CeedOperatorContextGetDouble(CeedOperator op, CeedContextFieldLabel field_label, size_t *num_values, const double **values);
+CEED_EXTERN int CeedOperatorContextGetInt32(CeedOperator op, CeedContextFieldLabel field_label, size_t *num_values, const int **values);
```
Comment (jrwrigh) [inline include/ceed/ceed.h:512]: Any reason to return `size_t *num_values`? The user would already need to ensure that `*values` is of the correct size, so they probably already know `num_values` (or could find out from the label itself).

Comment (jeremylt) [reply include/ceed/ceed.h:512]: I nearly always send the array length with the array so that each function call can stand alone

Comment (jrwrigh) [reply include/ceed/ceed.h:512]: Ah, didn't see that `values` was a double pointer. So my latter point is invalid (user doesn't have to know the length of the array before the call).

File: examples/fluids/src/turb_spanstats.c [hunk 2]
```diff
@@ -25,7 +25,7 @@ typedef struct {
   CeedElemRestriction elem_restr_parent_x, elem_restr_parent_stats, elem_restr_parent_qd, elem_restr_parent_colloc, elem_restr_child_colloc;
   CeedBasis           basis_x, basis_stats;
   CeedVector          x_coord, q_data;
-} * SpanStatsSetupData;
+} *SpanStatsSetupData;
```
Comment (jrwrigh) [inline examples/fluids/src/turb_spanstats.c:28]: Yeah, I didn't like that either, but `clang-format` wanted the extra space.

Comment (jeremylt) [reply examples/fluids/src/turb_spanstats.c]: Yeah. I should update to a newer `clang-format` when I get a chance. My local is ahead of the CI version

File: include/ceed/ceed.h [hunk 3]
```diff
@@ -509,7 +509,11 @@ CEED_EXTERN int CeedOperatorGetNumQuadraturePoints(CeedOperator op, CeedInt *num
 CEED_EXTERN int CeedOperatorGetFlopsEstimate(CeedOperator op, CeedSize *flops);
 CEED_EXTERN int CeedOperatorContextGetFieldLabel(CeedOperator op, const char *field_name, CeedContextFieldLabel *field_label);
 CEED_EXTERN int CeedOperatorContextSetDouble(CeedOperator op, CeedContextFieldLabel field_label, double *values);
+CEED_EXTERN int CeedOperatorContextGetDoubleRead(CeedOperator op, CeedContextFieldLabel field_label, size_t *num_values, const double **values);
+CEED_EXTERN int CeedOperatorContextRestoreDoubleRead(CeedOperator op, CeedContextFieldLabel field_label, const double **values);
```
Comment (jeremylt) [inline include/ceed/ceed.h:513]: I specified in the signature that this is read-only and added a restore function as well

**Review by jrwrigh — APPROVED** [review]
Looks good to me! Just one small question

File: interface/ceed-operator.c [hunk 4]
```diff
@@ -301,15 +416,15 @@ static int CeedOperatorContextSetGeneric(CeedOperator op, CeedContextFieldLabel
     CeedCall(CeedCompositeOperatorGetSubList(op, &sub_operators));
     if (num_sub != field_label->num_sub_labels) {
       // LCOV_EXCL_START
-      return CeedError(op->ceed, CEED_ERROR_UNSUPPORTED,
-                       "ContextLabel does not correspond to composite operator. Use CeedOperatorGetContextFieldLabel().");
+      return CeedError(op->ceed, CEED_ERROR_UNSUPPORTED, "ContextLabel does not correspond to composite operator");
```
[the remark this answers is not in the document]
Comment (jeremylt) [reply interface/ceed-operator.c:419]: Yeah, any context that doesn't have the field will leave a NULL value in this list of labels

**Pull request merged**